homeassistant/util/yaml.py

```diff
@@ -119,10 +120,15 @@ def _env_var_yaml(loader, node):
         raise HomeAssistantError(node.value)
 
 
+def _secret_yaml(loader, node):
+    """Load secrets and embed it into the configuration YAML."""
+    return secrets.get_secret(node.value) or ''
```

Comment:
I am also not sure why we would even make the rest of the app aware that we have a thing called secrets. Right now bootstrap is calling load secrets etc but we could do it on demand in this very method.

If we look at how we do it in `_include_yaml(…)`, we can have an auto secrets resolver with something like this:

```python
def _secret_yaml(loader, node):
    secret_path = os.path.join(os.path.dirname(loader.name), 'secrets.yaml')
    secrets = load_yaml(secret_path)
    if node.value not in secrets:
        raise HomeAssistantError('Specified secret {} not found'.format(node.value))
    return secrets[node.value]
```

Added bonus of this approach is that you can have a secret file per directory.

However, this is inefficient as we keep parsing the secret file every time we see a tag. To avoid that, we can create a cache on the loader.

```python
if not hasattr(loader, '_SECRET_CACHE'):
    loader._SECRET_CACHE = {}
if secret_path not in loader._SECRET_CACHE:
    if os.path.isfile(secret_path):
        loader._SECRET_CACHE[secret_path] = load_yaml(secret_path)
    else:
        loader._SECRET_CACHE[secret_path] = None
secrets = loader._SECRET_CACHE[secret_path]

if secrets is not None and node.value in secrets:
    return secrets[node.value]

# Handle keyring

# Throw exception if not found
```